### PR TITLE
Upgrade sbf tests to use bpf loader v3

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Problem
Follow up https://github.com/solana-labs/solana/pull/34194

Update programs/sbf/tests/programs.rs to use bpf_loader v3.

#### Summary of Changes

Fixes #

### CODE REVIEW

File: programs/sbf/tests/programs.rs [hunk 1]
```diff
@@ -518,6 +552,138 @@ fn test_sol_alloc_free_no_longer_deployable() {
...
+        mut genesis_config,
+        mint_keypair,
+        ..
+    } = create_genesis_config(50);
+
+    // deactivate `disable_deploy_of_alloc_free_syscall` feature so that the program
+    // that depends on sol_alloc_free_ can be deployed and executed.
+    genesis_config
+        .accounts
+        .remove(&solana_sdk::feature_set::disable_deploy_of_alloc_free_syscall::id());
```
Comment (pgarg66) [inline programs/sbf/tests/programs.rs:614]: It seems this feature is already activated on mainnet-beta. Do we want to keep the test where it's disabled?

File: programs/sbf/tests/programs.rs [hunk 2]
```diff
@@ -264,6 +264,37 @@ fn load_program_and_advance_slot(
     )
 }
 
+fn load_upgradeable_program_and_advance_slot(
```
Comment (pgarg66) [inline programs/sbf/tests/programs.rs:267]: Number of lines of code change will potentially reduce if `buffer_keypair` and `program_keypair` were constructed within this function, instead of being passed in as parameters.

File: programs/sbf/tests/programs.rs (hunk 1 above)
Comment (HaoranYi) [reply programs/sbf/tests/programs.rs:614]: I vote for no. Just keep `test_sol_alloc_free_no_longer_deployable_with_upgradeable_loader` and remove `test_sol_alloc_free_deployable_with_upgradeable_loader`.

Comment (HaoranYi) [reply programs/sbf/tests/programs.rs:614]: @Lichtso WDYT?

File: programs/sbf/tests/programs.rs [hunk 3]
```diff
@@ -848,33 +1005,49 @@ fn test_program_sbf_invoke_sanity() {
...
-            load_program(&bank_client, &bpf_loader::id(), &mint_keypair, program.1);
-        let invoked_program_id =
-            load_program(&bank_client, &bpf_loader::id(), &mint_keypair, program.2);
-        let (bank, noop_program_id) = load_program_and_advance_slot(
+        let authority_keypair = Keypair::new();
+
+        let invoke_buffer_keypair = Keypair::new();
+        let invoke_program_keypair = Keypair::new();
+        let invoke_program_id = invoke_program_keypair.pubkey();
+        load_upgradeable_program(
```
Comment (pgarg66) [inline programs/sbf/tests/programs.rs:1021]: Most of the new calls to `load_upgradeable_programs` take in buffer and program keypair as input parameters, but the two values are not used afterwards. We could add a wrapper to the function that generates these two keypairs internally, and just returns the pubkey of program.

Comment (HaoranYi) [reply programs/sbf/tests/programs.rs]: updated.

File: programs/sbf/tests/programs.rs (hunk 2 above)
Comment (HaoranYi) [reply programs/sbf/tests/programs.rs]: updated.

File: programs/sbf/tests/programs.rs (hunk 1 above)
Comment (HaoranYi) [reply programs/sbf/tests/programs.rs]: removed.

File: programs/sbf/tests/programs.rs [hunk 4]
```diff
@@ -3464,37 +3362,28 @@ fn test_program_sbf_realloc_invoke() {
         mut genesis_config,
         mint_keypair,
         ..
-    } = create_genesis_config(1_000_000_000_000);
+    } = create_genesis_config(100_000_000_000_000);
```
Comment (pgarg66) [inline programs/sbf/tests/programs.rs:3365]: why was this needed?

File: programs/sbf/tests/programs.rs [hunk 5]
```diff
@@ -2382,43 +2340,34 @@ fn test_program_sbf_invoke_upgradeable_via_cpi() {
...
-    load_upgradeable_program(
+    let program_id = load_upgradeable_program_wrapper(
         &bank_client,
         &mint_keypair,
-        &buffer_keypair,
-        &program_keypair,
         &authority_keypair,
         "solana_sbf_rust_upgradeable",
     );
+    bank_client.set_sysvar_for_tests(&clock::Clock {
```
Comment (pgarg66) [inline programs/sbf/tests/programs.rs:2367]: Is this needed due to delay_visibility?

**Comment by pgarg66** [comment]
Looks good to me!
Leaving it open to give @Lichtso some time to review it as well.

File: programs/sbf/tests/programs.rs (hunk 4 above)
Comment (HaoranYi) [reply programs/sbf/tests/programs.rs:3365]: not needed. reverted.

File: programs/sbf/tests/programs.rs (hunk 5 above)
Comment (HaoranYi) [reply programs/sbf/tests/programs.rs:2367]: yeah.

**Review by Lichtso — APPROVED** [review]
